src/modules/esl-utils/async/throttle.ts

```diff
@@ -26,24 +26,20 @@ export function throttle<F extends AnyToAnyFnSignature>(fn: F, threshold = 250,
 
   function throttledSubject(...args: any[]): void {
     const now = Date.now();
-
-    if (!last || now >= last + threshold) {
-      last = now;
-      fn.apply(this, args);
-    }
+    const lastThreshold = last + threshold;
 
     deferred = deferred || createDeferred();
+
     (typeof timeout === 'number') && clearTimeout(timeout);
     timeout = window.setTimeout(() => {
-      last = now;
+      last = Date.now();
       timeout = null;
-      // fn.apply to save call context
       deferred!.resolve(fn.apply(thisArg || this, args));
       deferred = null;
-    }, threshold);
+    }, (last || now < lastThreshold) ? lastThreshold - now : 0);
```

Comment:
The check `(last || now < lastThreshold)` looks a little bit complicated
So the only moment when it is not true is the initial call when `last` is not defined
but in that case then `now < lastThreashold` ~ ` now < undefined + treshold ` which is not true starting from at least 2th of January 1970

Other words it sounds like 
`(last || now < lastThreshold) ? lastThreshold - now : 0`
->
`(now < lastThreshold) ? lastThreshold - now : 0` (which is super logical and there is no questions for such check)
->
`Math.max(lastThreshold - now, 0)`